cql3/statements/modification_statement.cc

```diff
@@ -272,24 +270,42 @@ modification_statement::do_execute(query_processor& qp, service::query_state& qs
     _restrictions->validate_primary_key(options);
 
     if (has_conditions()) {
-        return execute_with_condition(qp, qs, options);
+        co_return co_await execute_with_condition(qp, qs, options);
     }
 
-    return execute_without_condition(qp, qs, options).then([] (coordinator_result<> res) {
-        if (!res) {
-            return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
-                    seastar::make_shared<cql_transport::messages::result_message::exception>(std::move(res).assume_error()));
+    json_cache_opt json_cache = maybe_prepare_json_cache(options);
+    std::vector<dht::partition_range> keys = build_partition_keys(options, json_cache);
+
+    if (keys.size() != 1) {
+        co_return ::shared_ptr<cql_transport::messages::result_message>{};
+    }
+
+    auto res = co_await execute_without_condition(qp, qs, options, json_cache, keys);
+    
+    if (!res) {
+        co_return seastar::make_shared<cql_transport::messages::result_message::exception>(std::move(res).assume_error());
+    }
+
+    auto result = seastar::make_shared<cql_transport::messages::result_message::void_message>();
+
+    auto&& table = s->table();
+    if (_may_use_token_aware_routing && table.uses_tablets()) {
+        auto token = keys[0].start()->value().token();
```

Comment:
keys[0] will be undefined behavior since keys is empty here (moved inside execute_without_condition()).

It would be clearer that this is happening if execute_without_condition() took it by value and we had std::move(keys) at the call site.

Reply:
Fixed and tests added